plan/plan.go

```diff
@@ -0,0 +1,75 @@
+package plan
+
+// DNSRecord holds information about a DNS record.
+type DNSRecord struct {
```

Comment:
in the project structure proposal it is in the plan package, but it might make sense to create a separate package for this, if this is something to be shared and used by other packages. but fine with me for now :)